org.osgi.util.converter/src/org/osgi/util/converter/ConvertingImpl.java

```diff
@@ -332,6 +344,42 @@ private <T> T convertToCollectionType(InternalConverter converter) {
 		return convertToCollection(converter);
 	}
 
+	@SuppressWarnings("unchecked")
+	private <T> T convertToOptionalType(InternalConverter converter) {
+
+		if (typeArguments != null) {
+			Object o = converter.convert(object).to(typeArguments[0]);
+			return (T) Optional.ofNullable(o);
+		}
+
+		return (T) Optional.ofNullable(object);
+
+	}
+
+	@SuppressWarnings("unchecked")
+	private <T> T convertToOptionalInt(InternalConverter converter) {
+
+		int i = converter.convert(object).to(int.class);
```

Comment:
There is unnecessary boxing here which could NPE.

```suggestion
		Integer i = converter.convert(object).to(Integer.class);
		return i == null ? OptionalInt.empty() : OptionalInt.of(i.intValue());
```